File: InJava/app/src/main/java/com/barryzeha/quizquestion/model/di/AppModule.java
package com.barryzeha.quizquestion.model.di;

import android.app.Application;

import androidx.room.Room;

import com.barryzeha.quizquestion.model.database.RankingDB;
import com.barryzeha.quizquestion.model.repository.QuestionRepository;
import com.barryzeha.quizquestion.model.repository.QuestionRepositoryImpl;
import com.barryzeha.quizquestion.model.repository.RankingDataSource;
import com.barryzeha.quizquestion.model.repository.RankingDataSourceImpl;
import com.barryzeha.quizquestion.model.repository.QuestionDataSource;
import com.barryzeha.quizquestion.model.repository.QuestionDataSourceImpl;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;

/****
 * Project QuizQuestion
 * Created by Barry Zea H. on 01/10/2022
 * Copyright (c)  All rights reserved.
 ***/
@Module
@InstallIn(SingletonComponent.class)
public class AppModule {
    @Provides
    @Singleton
    RankingDB databaseProvider(Application app){
        return Room.databaseBuilder(app, RankingDB.class,"ranking_table").build();
    }
    @Provides
    RankingDataSource localDataSourceProvider(RankingDB db) {return new RankingDataSourceImpl(db);}

    @Provides

    QuestionDataSource questionProvider(){ return new QuestionDataSourceImpl();}


}

File: InKotlin/app/src/main/java/com/barryzeha/quizquestion/di/AppModule.kt
package com.barryzeha.quizquestion.di

import android.app.Application
import androidx.room.Room
import com.barryzeha.quizquestion.model.database.RankingDB
import com.barryzeha.quizquestion.model.repository.LocalDataSource
import com.barryzeha.quizquestion.model.repository.LocalDataSourceImpl
import com.barryzeha.quizquestion.model.repository.QuestionRepositoryImpl
import com.barryzeha.quizquestion.model.repository.QuestionsRepository
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

/****
 * Project QuizQuestion
 * Created by Barry Zea H. on 29/09/2022
 * Copyright (c)  All rights reserved.
 ***/
@Module
@InstallIn(SingletonComponent::class)
class AppModule {

    @Provides
    @Singleton
    fun databaseProvider(app:Application)=Room.databaseBuilder(app, RankingDB::class.java,"ranking_table").build()

    @Provides
    fun localDataSourceProvider(db:RankingDB):LocalDataSource=LocalDataSourceImpl(db)
    @Provides
    fun questionRepository():QuestionsRepository=QuestionRepositoryImpl()

}